### PR TITLE
Make freeport testing friendly

### PR DESCRIPTION
**Pull request by dadgar**
This PR allows the caller to decide if they would like to have the
calling test fail, have the caller panic on error, or handle the errors
themselves.

### CODE REVIEW

File: lib/freeport/freeport.go [hunk 1]
```diff
@@ -72,16 +75,37 @@ func tcpAddr(ip string, port int) *net.TCPAddr {
...
+
+	return ports
+}
+
+// GetT is suitable for use when retrieving unused ports in tests. If there is
+// an error retrieving free ports, the test will be failed.
+func GetT(t testing.T, n int) (ports []int) {
+	ports, err := Free(n)
+	if err != nil {
+		panic(err)
```
Comment (slackpad) [inline lib/freeport/freeport.go]: did you mean to `t.Fatal` here?

Comment (dadgar) [reply lib/freeport/freeport.go]: Oops 😄

File: vendor/vendor.json [hunk 2]
```diff
@@ -2,94 +2,550 @@
 	"comment": "",
```
Comment (preetapan) [inline vendor/vendor.json]: You'll need to run make vendorfmt to fix formatting (our make dev also runs this for you(

Comment (dadgar) [reply vendor/vendor.json]: Made the change. Curious why you all do this?

Comment (preetapan) [reply vendor/vendor.json]: no more merge conflict hell, also its much easier to see and understand a diff at a glance, and it is sorted by path which is nice.

Comment (slackpad) [reply vendor/vendor.json]: good catch! yeah it makes merging the vendor file...possible - we have some extra stuff in the enterprise fork and it was a nightmare before `vendorfmt`

Comment (dadgar) [reply vendor/vendor.json]: Nice! Thanks for explaining 👍

**Pull request merged**